### DISCUSSION TITLE
System.InvalidOperationException: 'No service for type 'Quartz.Spi.ITypeLoadHelper' has been registered.'

### DISCUSSION THREAD

**Discussion by hubermarkus** [quartznet/quartznet#973] [General] [answered]
Hi, 
we are running a asp.net core 3.1 application. where  we added the following code:
public void ConfigureServices(IServiceCollection services)
{
...
            services.AddQuartz(q =>
            {
                // handy when part of cluster or you want to otherwise identify multiple schedulers
                q.SchedulerId = "Tunnel-Core-Scheduler";

                // we take this from appsettings.json, just show it's possible
                 q.SchedulerName = "Tunnel Core Scheduler";
            });

            // ASP.NET Core hosting
            services.AddQuartzServer(options =>
            {
                // when shutting down we want jobs to complete gracefully
                options.WaitForJobsToComplete = true;
            });
}

When we start the project i get "System.InvalidOperationException: 'No service for type 'Quartz.Spi.ITypeLoadHelper' has been registered.'"

what i am doing wrong?

**Comment by lahma** [comment, marked as answer]
You are missing type loader registration, which is:

```
 q.UseSimpleTypeLoader();
```

**Reply by hubermarkus** [reply]
great, that worked!
